### PR TITLE
bpo-24024: update str.__doc__ to show default encoding explicitly

### PR DESCRIPTION
**Pull request by plusminushalf**
sys.getdefaultencoding() always returns 'utf-8'
Reference: Objects/unicodeobject.c line number -> 4214

Also str() accepts bytes, bytearray or buffer-like objects as input
hence changing it to object.
As str returns .\_\_str\_\_() method defined inside object or repr(obj)
otherwise.

### CODE REVIEW

**Comment by plusminushalf** [comment]
```python
>>> str.__doc__
'str(object=\'\') -> str\nstr(object, encoding="utf-8", errors="strict") -> str\n\nCreate a new string object from the given object. If encoding or\nerrors is specified, then the object must expose a data buffer\nthat will be decoded using the given encoding and error handler.\nOtherwise, returns the result of object.__str__() (if defined)\nor repr(object).\nencoding defaults to sys.getdefaultencoding().\nerrors defaults to \'strict\'.'
```

File: Objects/unicodeobject.c
Comment (vadmium) [inline Objects/unicodeobject.c]: Best to use single quotes to be consistent with the line above.

Since you add the defaults to the signature, I think you can remove them from the end of the text below.

**Review by vadmium — COMMENTED** [review]
In general this looks worthwhile to me, just some minor suggestions

File: Objects/unicodeobject.c
Comment (DimitrisJim) [inline Objects/unicodeobject.c]: I'm not seeing why this needs two signatures now when `str(object='', encoding='utf-8', errors='strict')` would suffice.

Comment (garvitdelhi) [inline Objects/unicodeobject.c]: I am not seeing two signatures? This just have one signature that is ```str(object, encoding='utf-8', errors='strict') ```

File: Objects/unicodeobject.c
Comment (vadmium) [inline Objects/unicodeobject.c]: Passing no arguments is also supported:

```python console
>>> str()
''
```

Comment (DimitrisJim) [reply Objects/unicodeobject.c]: Indeed, object needs a default value, i.e `object=''`.

**Review by DimitrisJim — APPROVED** [review]

File: Objects/unicodeobject.c
Comment (serhiy-storchaka) [inline Objects/unicodeobject.c:15122]: There is a problem with this. This signature means that `str('', 'utf-8', 'strict')` is equivalent to `str()`. But calling `str` with string argument and encoding is an error. The signature of `str` can't be expressed in such form, this is a cause why the `str` constructor still is not converted to Argument Clinic.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Sorry, it was approved by mistake. I don't approve this patch in the current form.

File: Objects/unicodeobject.c
Comment (merwok) [reply Objects/unicodeobject.c]: I don’t think that merging the two lines is correct.

**Comment by brettcannon** [comment]
To try and help move older pull requests forward, we are going through and backfilling 'awaiting' labels on pull requests that are lacking the label. Based on the current reviews, the best we can tell in an automated fashion is that a core developer requested changes to be made to this pull request.

If/when the requested changes have been made, please leave a comment that says, `I have made the requested changes; please review again`. That will trigger a bot to flag this pull request as ready for a follow-up review.

File: Objects/unicodeobject.c
Comment (merwok) [reply Objects/unicodeobject.c:15122]: I don’t think that merging the two lines is correct.

**Comment by serhiy-storchaka** [comment]
Closed because it make the docstring incorrect.

**Pull request closed without merging**
